codex-rs/exec/src/lib.rs

```diff
@@ -188,6 +188,17 @@ pub async fn run_main(cli: Cli, codex_linux_sandbox_exe: Option<PathBuf>) -> any
         std::process::exit(1);
     }
 
+    // If inside a git repo but with no commits, provide a clearer message.
+    if !skip_git_repo_check
+        && get_git_repo_root(&config.cwd.to_path_buf()).is_some()
```

Comment:
Can you unify this with the code on 186 so it's clear we only run `get_git_repo_root()` once?

An option is to change `get_git_repo_root()` to return an `enum` with three states: Yes, YesButNoCommits, No, though I don't know if that's really necessary if we consolidate this code with the above and change `git_repo_has_commits()` to invoke `git` once.